numba/tests/test_parallel_backend.py

```diff
@@ -690,6 +690,42 @@ def main():
             self.assertIn("Terminating: Nested parallel kernel launch detected",
                           e_msg)
 
+    def test_workqueue_handles_fork_from_non_main_thread(self):
+        # For context see #7872, but essentially the multiprocessing pool
+        # implementation has a number of Python threads for handling the worker
+        # processes, one of which calls fork(2), this results in a fork from a
+        # non-main thread.
+
+        runme = """if 1:
+            from numba import njit, prange, threading_layer
+            import numpy as np
+            import multiprocessing
+
+            if __name__ == "__main__":
+
+                @njit(parallel=True)
+                def func(x):
+                    return 10. * x
+
+                arr = np.arange(2.)
+
+                # run in single process to start Numba's thread pool
+                np.testing.assert_allclose(func(arr), func.py_func(arr))
+
+                # now run in a multiprocessing pool to get a fork from a
+                # non-main thread
+                with multiprocessing.Pool(10) as p:
```

Comment:
needs to force a fork context here

Reply:
Good point, OSX defaults to `'spawn'`.

Reply:
b07d687 fixes, also adds a test skip for OS which don't have `fork(2)` and adds an assert to ensure the expected threading layer was used.